internal/wallet/utxo_test.go

```diff
@@ -0,0 +1,153 @@
+package wallet
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/btcsuite/btcwallet/wtxmgr"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	// Spends: bogus
+	// Outputs: 10 BTC
+	fakeTxRecordA *wtxmgr.TxRecord
+
+	// Spends: A:0
+	// Outputs: 5 BTC, 5 BTC
+	fakeTxRecordB *wtxmgr.TxRecord
+
+	exampleBlock100 = makeBlockMeta(100)
+)
+
+// Test setup?
+// 		create wallet
+// 		mine regtest coins?
+//  	ListUnspent() to check if we can see the mined coins?
+
+// TestListUnspent() will also need to check different types of scripts
+func TestListUnspent(t *testing.T) {
+	tearDownFunc, wallet := setupWallet(t)
+	defer tearDownFunc()
+
+	utxos := fakeUtxos(wallet)
+
+	syncBlock := wallet.manager.SyncedTo()
+
+	err := walletdb.View(wallet.db, func(tx walletdb.ReadTx) error {
+		wtxmgrBucket := tx.ReadBucket(wtxmgrNamespaceKey)
+		if wtxmgrBucket == nil {
+			return errors.New("missing transaction manager namespace")
+		}
+		result := wallet.credit2ListUnspentResult(utxos[0], syncBlock, wtxmgrBucket)
+
+		assert.NotNil(t, result)
+		return nil
+	})
+	if err != nil {
+		fmt.Println(err)
+	}
+	assert.Nil(t, err)
+
+}
+
+// fakeUtxos creates fake transactions, and inserts them into the provided wallet's db
+func fakeUtxos(w *wallet) []wtxmgr.Credit {
```

Comment:
Maybe these functions  (`fakeUtxos, spendOutput, makeBlockMeta`) should be moved to `test_util.go` ?

Reply:
moved